### ISSUE TITLE
Help: Update interval not matching update times

### ISSUE THREAD

**Issue by jsilvagi** [qdm12/ddns-updater#108]
**TLDR**: Updates upon startup, does not do periodic update, no matter the interval.

<img width="337" alt="Screen Shot 2020-10-05 at 12 26 37 PM" src="https://user-images.githubusercontent.com/29311129/95106086-0cbf6480-0706-11eb-90c2-1b403aa6fe6b.png">

I assume the time at the end is the time it last sent an update?  The only time it resets is after I restart the container and it does not update on that interval.  I have tried mins, hours, and even leaving it blank.  Any thoughts?

1. Is this urgent?

    - [ ] Yes
    - [x] No

2. What DNS service provider(s) are you using?

    - [ ] Cloudflare
    - [ ] DDNSS.de
    - [ ] DonDominio
    - [ ] DNSPod
    - [ ] Dreamhost
    - [ ] DuckDNS
    - [ ] DynDNS
    - [ ] GoDaddy
    - [ ] Google
    - [ ] He.net
    - [ ] Infomaniak
    - [ ] Namecheap
    - [x] NoIP

3. What's the version of the program?

Running version latest built on 2020-10-04T14:38:55Z (commit e630dd9)

4. What are you using to run the container?

    - [ ] Docker run
    - [x] Docker Compose
    - [ ] Kubernetes
    - [ ] Docker stack
    - [ ] Docker swarm
    - [ ] Podman
    - [ ] Other:

5. Extra information

Logs:

```

2020-10-05T15:08:36.733Z	INFO	Found single setting to update record


2020-10-05T15:08:37.135Z	INFO	Reading history from database: domain XXXX host @


2020-10-05T15:08:37.135Z	INFO	Web UI listening at address 0.0.0.0:8000 with root URL ""


2020-10-05T15:08:37.136Z	INFO	backup: disabled


2020-10-05T15:09:07.842Z	INFO	http server: HTTP GET /


2020-10-05T15:09:09.499Z	INFO	http server: HTTP GET /


2020-10-05T15:11:54.754Z	INFO	http server: HTTP GET /


2020-10-05T15:16:23.499Z	INFO	http server: HTTP GET /

```

Configuration file:

```
  ddns-updater:
    image: qmcgaw/ddns-updater
    container_name: ddns-updater
    network_mode: bridge
    ports:
      - 8000:8000/tcp
    volumes:
      - ${USERDIR}/docker/ddns-updater/data:/updater/data
    environment:
#      - CONFIG=
      - PERIOD=5m
      - IP_METHOD=cycle
      - IPV4_METHOD=cycle
      - IPV6_METHOD=cycle
      - HTTP_TIMEOUT=10s

      # Web UI
      - LISTENING_PORT=8000
      - ROOT_URL=/

      # Backup
      - BACKUP_PERIOD=0 # 0 to disable
      - BACKUP_DIRECTORY=/updater/data

      # Other
      - LOG_ENCODING=console
      - LOG_LEVEL=info
      - NODE_ID=-1 # -1 to disable
      - GOTIFY_URL=
      - GOTIFY_TOKEN=
    restart: always
    labels:
      - com.centurylinklabs.watchtower.enable=true


{
    "settings": [
        {
            "provider": "noip",
            "domain": "XXX",
            "host": "@",
            "username": "XXX",
            "password": "XXX"
        }
    ]
}


```

Host OS:

Ubuntu 20.04.1 LTS

**Comment by qdm12** [comment]
Thanks for the detailed issue!

By default, it does a DNS lookup to compare the current IP address for your domain with your public IP address. It does check your public IP address periodically though. If these mismatch, then an update is made. Otherwise, it waits for the next period to recheck your public IP address and do a DNS lookup of your domain.

This is so to limit the number of calls to the DNS providers as some of them block you if you send too many requests.

I don't recall exactly why the first run sends an update request, that should probably be changed to follow the same logic. Let's keep the issue opened for that one.

What you can try is to manually change your A record to 127.0.0.1 and wait to see if it gets updated by the container (it should I hope haha). Let me know of I missed out something else.

**Comment by jsilvagi** [comment]
That covers it, thank you for the quick response.  I figured it was working as intended, just the time was confusing.  I am using this for a remote location that I won't have easy access to, so I wanted to make sure it was all working correctly.

Thanks for this app, was a perfect solution for me!

**Comment by qdm12** [comment]
🎉  Thanks for the donation! I'm also running that container at a "remote location that I won't have easy access to", also known as mother's place on another continent haha! So I feel you...

But it has been keeping my DNS records updated so far. Just let me know in case you use IPv6, the container should work but it's not thoroughly tested it for it.

**Comment by jsilvagi** [comment]
No problem, well earned!  This has solved a big issue for me.  I have had it running for a couple weeks already and it has been working fine, but an internet outage made me look at things deeper and wanted to understand how the updates occurred.  Thanks again!

**Comment by qdm12** [comment]
Hey I just wanted to check if it has been working fine so far? 🤔 

I also checked the logic, and it is supposed to update your records at start only if the IP address changed from either the DNS lookup or the persisted data (if `no_dns_lookup` is specified).

**Comment by jsilvagi** [comment]
It has been working great, thanks!

**Comment by qdm12** [comment]
Thanks for letting me know! Enjoy!

PS: I'm working on a new UI, make sure to follow github releases 😉
